meta-mender-raspberrypi-demo/conf/layer.conf

```diff
@@ -0,0 +1,19 @@
+# Board specific layer configuration for meta-mender
+# Copyright 2018 Northern.tech AS
+
+# We have a conf and classes directory, add to BBPATH
+BBPATH .= ":${LAYERDIR}"
+
+# We have recipes-* directories, add to BBFILES
+BBFILES += "${LAYERDIR}/recipes-*/*/*.bb \
+            ${LAYERDIR}/recipes-*/*/*.bbappend"
+
+BBFILE_COLLECTIONS += "mender-raspberrypi-demo"
+BBFILE_PATTERN_mender-raspberrypi-demo = "^${LAYERDIR}/"
+# Priority is one lower than that of the mender-raspberrypi layer, so that the wifi append is applied after
+# the mender-raspberrypi append file.
+BBFILE_PRIORITY_mender-raspberrypi-demo = "9"
```

Comment:
higher number = higher priority 

So if you have lower priority numer then then mender-raspberrypi in this layer, the meta-rasberrypi layer changes will be applied after this one. Which does not match your comment. 

If you need the changes of meta-raspberrypi-demo to be applied "last" then it has to have the highest priority (number), e.g. 11 which is one above meta-mender-raspberrypi